patch.js

```diff
@@ -1,41 +1,114 @@
 const fs = require('node:fs');
 const crypto = require('node:crypto');
 const path = require('node:path');
-const { pki, md } = require('node-forge');
+const { asn1, pki, md } = require('node-forge');
 const prompt = require('prompt');
 const colors = require('@colors/colors/safe');
+const dotenv = require('dotenv');
+const { program } = require('commander');
+
+const optionsConfig = {
+	nintendo_ca_g3_path: {
+		shortOption: 'g3',
+		default: './CACERT_NINTENDO_CA_G3.der',
+		description: 'Path to Nintendo CA - G3 certificate (may be in DER or PEM format, default to this directory)',
+		env: 'SSSL_NINTENDO_CA_G3_PATH'
+	},
+	nintendo_ca_g3_format: {
+		shortOption: 'f',
+		default: 'der',
+		description: 'Nintendo CA - G3 certificate format (must be "der" or "pem")',
+		env: 'SSSL_NINTENDO_CA_G3_FORMAT'
+	},
+	ca_private_key_path: {
+		shortOption: 'cap',
+		default: undefined,
+		description: 'Path to private key for forged CA (will generate if not set)',
+		env: 'SSSL_CA_PRIVATE_KEY_PATH'
+	},
+	site_private_key_path: {
+		shortOption: 'sp',
+		default: undefined,
+		description: 'Path to private key for site certificate (will generate if not set)',
+		env: 'SSSL_SITE_PRIVATE_KEY_PATH'
+	},
+	csr_path: {
+		shortOption: 'csrp',
+		default: undefined,
+		description: 'Path to CSR (will generate if not set)',
+		env: 'SSSL_CSR_PATH'
+	},
+	common_name: {
+		shortOption: 'cn',
+		default: '*',
+		description: 'CN for site certificate (default to "*")',
+		env: 'SSSL_COMMON_NAME'
+	},
+	output_folder_path: {
+		shortOption: 'o',
+		default: './',
+		description: 'Output folder (default to this directory)',
+		env: 'SSSL_OUTPUT_FOLDER_PATH'
+	}
+};
+
+async function main() {
+	dotenv.config();
+
+	program.option('-i, --interactive', 'Interactively prompt for all configuration values');
+	for (const [option, config] of Object.entries(optionsConfig)) {
+		program.option(`-${config.shortOption}, --${option} <value>`, config.description);
```

Comment:
Not an issue per-say, but wouldn't this make the options use underscores in their long forms rather than the typical hyphens? Is that intended?

Reply:
Yes. It's not really intentional, but it makes things easier since commander does some case transformation to dashed arguments from kebab to camel. So, using using dashed arguments would require another step to transform the case back.

I'm not entirely sure why the options use  snake case anyway while most other variables use camel case. If all the options were renamed to use camel case, it would be easier to use kebab case for arguments.

Reply:
> I'm not entirely sure why the options use snake case anyway while most other variables use camel case

It's entirely a preference on my part. For variables I always use camelCase, and in JSON I use snake_case

> Yes. It's not really intentional, but it makes things easier since commander does some case transformation to dashed arguments from kebab to camel. So, using using dashed arguments would require another step to transform the case back.

It's undocumented, but this can easily be worked around without extra processing after the fact. Commander provides a `addOption` method which lets you pass in a raw `Option` class. It's designed for use cases like this. By using a custom `Option` class and overriding the undocumented `attributeName` method, the camelCase conversion can be removed entirely:

```js
const { program, Option } = require('commander');

class CustomOption extends Option {
	attributeName() {
		// "this.name().replace(/^no-/, '')" is from commander
		// "replace(/-/g, '_')" converts test-argument to test_argument
		return this.name().replace(/^no-/, '').replace(/-/g, '_');
	}
}

program.addOption(new CustomOption('-t, --test-argument', 'Test argument'));

program.parse();

const options = program.opts();

console.log(options); // { test_argument: true }
```